src/dstack/_internal/core/models/fleets.py

```diff
@@ -0,0 +1,199 @@
+import ipaddress
+from datetime import datetime
+from enum import Enum
+from typing import Any, Dict, List, Optional, Type, Union
+
+from pydantic import Field, root_validator, validator
+from typing_extensions import Annotated, Literal
+
+from dstack._internal.core.models.backends.base import BackendType
+from dstack._internal.core.models.common import CoreModel
+from dstack._internal.core.models.instances import SSHKey
+from dstack._internal.core.models.pools import Instance
+from dstack._internal.core.models.profiles import (
+    DEFAULT_POOL_TERMINATION_IDLE_TIME,
+    Profile,
+    ProfileParams,
+    ProfileRetry,
+    SpotPolicy,
+    TerminationPolicy,
+    parse_duration,
+)
+from dstack._internal.core.models.resources import Range, ResourcesSpec
+
+
+class FleetStatus(str, Enum):
+    # Currently all fleets are ACTIVE/TERMINATING/TERMINATED
+    # SUBMITTED/FAILED may be used if fleets require async processing
+    SUBMITTED = "submitted"
+    ACTIVE = "active"
+    TERMINATING = "terminating"
+    TERMINATED = "terminated"
+    FAILED = "failed"
+
+
+class InstanceGroupPlacement(str, Enum):
+    ANY = "any"
+    CLUSTER = "cluster"
+
+
+class SSHHostParams(CoreModel):
+    hostname: Annotated[str, Field(description="The IP address or domain to connect to")]
+    port: Annotated[
+        Optional[int], Field(description="The SSH port to connect to for this host")
+    ] = None
+    user: Annotated[Optional[str], Field(description="The user to log in with for this host")] = (
+        None
+    )
+    ssh_key_path: Annotated[
+        Optional[str], Field(description="The private key to use for this host")
+    ] = None
+    ssh_key: Optional[SSHKey] = None
+
+
+class SSHParams(CoreModel):
```

Comment:
@jvstme `region` for `dstack pool add-ssh` can be used to group interconnected/non-interconnected instances. This doesn't seem to be necessary for fleets since instances are grouped by fleets. Why would one need to add ssh fleets and specify different regions?

Reply:
If each instance has a region, it will allow to group instances across fleets or differentiate instances within a fleet based on their geographic location.

**Problem 1**: I have 2 fleets in Europe and 2 fleets in the US. I need to run a configuration on any of the fleets, as long as it is in the US.
**Solution 1**: Specify `region` in the run configuration and allow `dstack` to choose one of the US fleets automatically.

**Problem 2**: I need to run a service with 3 replicas: 1 in Europe, 1 in the US, 1 in Asia. I also need to have one stand-by instance in each region in case the other instance in this region fails.
**Solution 2**: Create the `my-fleet` fleet with 6 instances: 2 in Europe, 2 in the US, 2 in Asia. In the service configuration, specify `fleet: my-fleet`, `replicas: 3`, `placement: geo-distributed`.